R/lockbox.R

```diff
@@ -70,13 +52,30 @@ lockbox.list <- function(lock, env) {
   })
 }
 
-#' @export
-lockbox.default <- function(obj) {
-  stop(
-    "Invalid parameters passed to ", sQuote("lockbox"), " method: ",
-    "must be a ", sQuote("character"), " or ", sQuote("list"), " but ",
-    "instead I got a ", sQuote(class(obj)[1]), "."
-  )
+
+parse_lock <- function(lock, env = getOption("lockbox.env", "!packages")) {
+  if (is.character(lock) && length(lock) == 1) {
+    handle_float_version <- function(x) {
+      if (is.numeric(x)) { sprintf("%.1f", x) }
```

Comment:
Won't this fail on versions like `0.11`? I think we just want `as.character(x)` or a manual strsplit on the `.` if it exists.

```r
R-ENGINES> sprintf("%.1f", 0.11)
[1] "0.1"
```

Reply:
Wouldn't `x` always be numeric given that it's the float yml handler?

Reply:
You mentioned strsplit on `.` would we also need it for `-`?  Is there actually a limited character set that CRAN enforces for versions?

Reply:
@russellpierce This logic should only apply to yml values detected as being float values, e.g.,

```yml
version: 1.0
```

rather than

```yml
version: "1.0"
```